File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py

def solution(s):
    answer = True
    answerq = [] #(만담자
    
    for i in range(len(s)):
        if len(answerq) == 0 and s[i]=="(" :
            answerq.append(s[i])
        elif len(answerq) == 0 and s[i]==")" :
            answer=False
            break
            
        elif s[i] == "(" :
            answerq.append(s[i])
        
        elif s[i] == ")" :
            answerq.pop()
            
    if len(answerq) != 0 :
        answer = False
        

    
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        
        Stack<Character> st = new Stack<>();
        
        for (int i = 0 ; i<s.length(); i++)
        {
            char nowval = s.charAt(i);
            
            if (nowval == '(')
            {
                st.push(nowval);
            } else {
                if (st.isEmpty())
                {
                    return false;
                } else {
                    st.pop();
                }
            }
        }
        
        boolean answer = st.isEmpty() ? true : false;


        return answer;
    }
}